prov/shm/src/smr_ep.c

```diff
@@ -634,13 +634,13 @@ static int smr_format_sar(struct smr_ep *ep, struct smr_cmd *cmd,
 	return 0;
 }
 
-int smr_select_proto(bool use_ipc, bool cma_avail,
-		     uint32_t op, uint64_t total_len, uint64_t op_flags)
+int smr_select_proto(enum fi_hmem_iface iface, bool use_ipc, bool cma_avail,
+                     uint32_t op, uint64_t total_len, uint64_t op_flags)
 {
 	if (op == ofi_op_read_req) {
 		if (use_ipc)
 			return smr_src_ipc;
-		if (cma_avail && FI_HMEM_SYSTEM)
```

Comment:
Woah, that's a big miss. Can you add something in the commit message about how this fixes a bug where CMA would never be selected for a read?

Reply:
Thanks. Updated commit message. I will backport the change to 1.18.x branch.

Do we need it for 1.17.x as well?

Reply:
Well the only real use case of shm is efa so I'll let you decide how far back you want to go. The issue technically goes all the way back to 1.15.x

Reply:
Cool then we only need to backport to 1.18.x